components/butterfly/src/server.rs

```diff
@@ -447,7 +448,7 @@ impl Server {
                 let mut store = incarnation_store::IncarnationStore::new(path.join("INCARNATION"));
                 store.initialize()?;
 
-                let mut me = self.member.write().expect("Member lock is poisoned");
+                let mut me = self.member.write();
```

Comment:
There's an opportunity here to simplify this code and `sync_incarnation` considerably. Since this is the only code path for calling `sync_incarnation`; we can remove the need for the `match` inside it if we take a non-optional `IncarnationStore` like:
```rust
    fn sync_incarnation(&mut self, store: incarnation_store::IncarnationStore) -> Result<()> {
        let value = store.load()?;
        self.incarnation_store = Some(store);
        self.member.incarnation = value;
        INCARNATION.set(value.to_i64());
        debug!("Setting incarnation number to {}", self.member.incarnation);
        Ok(())
    }
```
(note: the doc comment would need updating)

Then, this call can become:
```suggestion
                self.member.write().sync_incarnation(store)?;
```
and the next two lines can be removed.

Reply:
Nice, that does look better.

Reply:
Nice.